frontend/integration-tests/tests/kubevirt/utils/utils.ts

```diff
@@ -0,0 +1,191 @@
+/* eslint-disable no-unused-vars, no-undef */
+import * as _ from 'lodash';
+import { execSync } from 'child_process';
+import { $, by, ElementFinder, browser, ExpectedConditions as until } from 'protractor';
+
+import { config } from '../../../protractor.conf';
+import { nameInput as loginNameInput, passwordInput as loginPasswordInput, submitButton as loginSubmitButton } from '../../../views/login.view';
+import { PAGE_LOAD_TIMEOUT } from './consts';
+
+
+export function removeLeakedResources(leakedResources: Set<string>) {
+  const leakedArray: Array<string> = [...leakedResources];
+  if (leakedArray.length > 0) {
+    console.error(`Leaked ${leakedArray.join()}`);
+    leakedArray.map(r => JSON.parse(r) as {name: string, namespace: string, kind: string})
+      .forEach(({name, namespace, kind}) => {
+        try {
+          execSync(`kubectl delete -n ${namespace} --cascade ${kind} ${name}`);
+        } catch (error) {
+          console.error(`Failed to delete ${kind} ${name}:\n${error}`);
+        }
+      });
+  }
+  leakedResources.clear();
+}
+
+export function addLeakableResource(leakedResources: Set<string>, resource) {
+  leakedResources.add(
+    JSON.stringify({name: resource.metadata.name, namespace: resource.metadata.namespace, kind: resource.kind})
+  );
+}
+
+export function removeLeakableResource(leakedResources: Set<string>, resource) {
+  leakedResources.delete(
+    JSON.stringify({name: resource.metadata.name, namespace: resource.metadata.namespace, kind: resource.kind})
+  );
+}
+
+export function createResource(resource) {
+  execSync(`echo '${JSON.stringify(resource)}' | kubectl create -f -`);
+}
+
+export function createResources(resources) {
+  resources.forEach(createResource);
+}
+
+export function deleteResource(resource) {
+  const kind = resource.kind === 'NetworkAttachmentDefinition' ? 'net-attach-def' : resource.kind;
+  execSync(`kubectl delete -n ${resource.metadata.namespace} --cascade ${kind} ${resource.metadata.name}`);
+}
+
+export function deleteResources(resources) {
+  resources.forEach(deleteResource);
+}
+
+export async function withResource(resourceSet: Set<string>, resource: any, callback: Function, keepResource: boolean = false) {
+  addLeakableResource(resourceSet, resource);
+  await callback();
+  if (!keepResource) {
+    deleteResource(resource);
+    removeLeakableResource(resourceSet, resource);
+  }
+}
+
+export async function click(elem: ElementFinder, timeout?: number, conditionFunc?: Function) {
+  const _timeout = timeout !== undefined ? timeout : config.jasmineNodeOpts.defaultTimeoutInterval;
+  if (conditionFunc === undefined) {
+    await browser.wait(until.elementToBeClickable(elem), _timeout);
+    await elem.click();
+  } else {
+    do {
+      await browser.wait(until.elementToBeClickable(elem), _timeout);
+      await elem.click();
+    } while (await conditionFunc() === false);
```

Comment:
This does not seem like a good pattern. Does this just sit forever if the element never appears?